lib/puppet_x/coi/jboss/internal/cli_executor.rb

```diff
@@ -0,0 +1,190 @@
+# Class that will handle executions of commands
+class Puppet_X::Coi::Jboss::Internal::CliExecutor
+  # Constructor
+  # @param {Puppet_X::Coi::Jboss::Internal::ExecutionStateWrapper} execution_state_wrapper handles command execution
+  def initialize(execution_state_wrapper)
+    @execution_state_wrapper = execution_state_wrapper
+    @evaluator = Puppet_X::Coi::Jboss::Internal::Evaluator.new
+  end
+
+  attr_writer :execution_state_wrapper
+
+  # Method that allows us to setup shell executor, used in tests
+  def shell_executor=(shell_executor)
+    @execution_state_wrapper.shell_executor = shell_executor
+  end
+
+  def shell_executor
+    @execution_state_wrapper.shell_executor
+  end
+
+  # Method that executes command, if method fails it prints log message
+  # @param {String} typename name of resource
+  # @param {String} cmd command that will be executed
+  # @param {String} way bring up|bring down to for logging
+  # @param {Hash} resource standard puppet resource object
+  def executeWithFail(typename, cmd, way, resource)
+    executed = wrap_execution(cmd, resource)
+    unless executed[:result]
+      ex = "\n#{typename} failed #{way}:\n[CLI command]: #{executed[:cmd]}\n[Error message]: #{executed[:lines]}"
+      unless $add_log.nil? and $add_log > 0
+        ex = "#{ex}\n#{printlog $add_log}"
+      end
+      raise ex
+    end
+    executed
+  end
+  # Method that executes command and returns outut
+  # @param {String} cmd command that will be executed
+  # @param {Boolean} runasdomain if command will be executen in comain instance
+  # @param {Hash} ctrlcfg hash with configuration
+  # @param {Number} retry_count number of retry after failed command
+  # @param {Number} retry_timeout timeout after failed command
+  def executeAndGet(cmd, runasdomain, ctrlcfg, retry_count, retry_timeout)
+    ret = run_command(cmd, runasdomain, ctrlcfg, retry_count, retry_timeout)
+    unless ret[:result]
+      return {
+        :result => false,
+        :data => ret[:lines]
+      }
+    end
+
+    # JBoss expression and Long value handling
+    ret[:lines].gsub!(/expression \"(.+)\",/, '\'\1\',')
+    ret[:lines].gsub!(/=> (\d+)L/, '=> \1')
+
+    begin
+      evaluated_output = @evaluator.evaluate(ret[:lines])
```

Comment:
Change name to not confuse with Eval - for ex.: `JBossOutputSanitizer#sanitizeOutput`